### PR TITLE
minor cleanups for the new p5 code, reduce options

### PR DESCRIPTION
**Pull request by ahnitz**
Requires #3829

### CODE REVIEW

File: bin/bank/pycbc_geom_aligned_bank [hunk 1]
```diff
@@ -563,14 +563,8 @@ else:
...
 num_banks = int((len(newV1s) - 0.5)//opts.split_bank_num) + 1
-if not opts.is_sub_workflow:
-    input_h5file = wf.File.from_path(opts.intermediate_data_file)
-else:
-    # Basically, if this is a sub-workflow do not store a PFN, as it will be
-    # some meaningless temporary path, but instead tell the uber-workflow
-    # that these files are outputs and used within the sub-dax and let it
-    # handle the rest.
-    input_h5file = pwf.File(os.path.basename(opts.intermediate_data_file))
+input_h5file = wf.File.from_path(opts.intermediate_data_file)
```
[the remark this answers is not in the document]
Comment (spxiwh) [reply bin/bank/pycbc_geom_aligned_bank:566]: The cleanup issue should not be a concern. Pegasus knows [with these changes] that the subworkflow is using these files and will not try to clean them up (same as with other input files). This does require the top-level dax to know about this, but it needs to know that anyway, or things don't work  (i.e. you cannot create a subworkflow which is generated during the dax, but will decide at generation time what files to use ... You have to declare that up front (or tell pegasus that it's using everything)).

The "bad" PFNs is indeed the problem I intended to solve, and while this was in flux on the pegasus side, I was seeing cases where the bad PFN was preferred (which is part of the reason this was the way it was). I guess this then stabilised on the pegasus end, and having the "bad" PFN isn't a problem. I still don't like having an invalid PFN in the workflow, but if it makes code more complicated elsewhere, I won't hold this up.

File: pycbc/workflow/core.py [hunk 2]
```diff
@@ -831,6 +810,13 @@ def save(self, filename=None, output_map_path=None):
         with open(ini_file, 'w') as fp:
             self.cp.write(fp)
 
+        # save the sites file
+        #FIXME change to just check for submit_now if we drop pycbc_submit_dax
```
[the remark this answers is not in the document]
Comment (spxiwh) [reply pycbc/workflow/core.py]: Pegasus-plan *can* take a site catalog (I had that running originally as a one-site-catalog-for-all-workflows thing, and then split it into a make-one-for-all-subworkflows). It works something like this:
https://github.com/gwastro/pycbc/pull/3594/commits/1454028071c3383f04e0f9a8c7a5e050e94735a6#diff-b801e87f9ea1189a32fb54aae82de070e4351a3bd631f8e04c49827b1dac1c2aR314
and can also be done for the `.plan` method.

File: pycbc/workflow/pegasus_workflow.py [hunk 3]
```diff
@@ -597,11 +583,11 @@ def save(self, filename=None, submit_now=False, plan_now=False,
                     # There was no storage path
                     pass
 
-        if (submit_now or plan_now) and self._asdag is None:
-            self.plan_and_submit(submit_now=submit_now)
-        else:
-            self._adag.write(filename)
-            if self._asdag is None:
+        self._adag.write(filename)
```
[the remark this answers is not in the document]
Comment (spxiwh) [reply pycbc/workflow/pegasus_workflow.py]: I poked around at the pegasus source code. I think that we *can* keep the `.dax` filename if we want, by doing exactly what you had here, and this change should *not* have resulted in a `.dax` and a `.yml` file. I think if the dax is written once, pegasus then uses that when planning (but would need testing)

**Review by spxiwh — APPROVED** [review]
I won't hold this up longer. I made a few comments in line, and accept that there may be some "badness" by doing this, but it's probably less than the "badness" you remove by doing this :-)

File: pycbc/workflow/core.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahnitz) [reply pycbc/workflow/core.py]: Oh, I see, it essentially occurs by means of overriding the pegasus properties.

**Pull request merged**